### PR TITLE
add stellar client

### PR DESCRIPTION
**Pull request by AlaaElattar**

### CODE REVIEW

File: packages/stellar_client/lib/src/client.dart [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+      await sdk.accounts.account(keyPair.accountId);
+      return keyPair;
+    } catch (error) {
+      print("Failed to create account $error");
+    }
+    return null;
+  }
+
+  // If testnet account activated with FriendBot, the asset should be added manually.
+  Future<bool> activateTestNetAccount({required String accountId}) async {
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:119]:
```suggestion
  Future<bool> activateThroughFriendBot({required String accountId}) async {
```

File: packages/stellar_client/lib/src/client.dart [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+    try {
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:120]: please check there that the network selected is testnet and throw an error if it's public

File: packages/stellar_client/lib/src/client.dart [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+
+      transaction.sign(sourceKeyPair, Network.TESTNET);
+      await sdk.submitTransaction(transaction);
+      print("Transaction successful.");
+    } catch (error) {
+      print("Failed to send transaction: $error");
+    }
+  }
+
+  Future<Transaction?> getActivationTransaction() async {
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:217]: please make this method private, the user shouldn't have access to it

File: packages/stellar_client/lib/src/client.dart [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+        body: jsonEncode({'transaction': fundedTransaction.toXdr()}),
+      );
+
+      print(response.body);
+    } catch (error) {
+      throw Exception('Something went wrong! $error');
+    }
+  }
+
+  Future<void> getTransactions() async {
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:395]: this method should return a list of transactions

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
The current implementation combines a service and a client. Part of it asks for the keypair to use for sending a transaction and the other part uses the loaded one. I believe we should load the account on the constructor, then we can use the loaded account to activate the account, send transactions, get balance, list transactions, ... and so on. 
For creating an account, it can be a factory constructor that generates a keypair and returns a client instance.

```ts
  // create new account
  factory Client.create(network) {
    final keyPair = KeyPair.random();
    return Client(network, keyPair.secretSeed);
  }
```

File: packages/stellar_client/lib/src/client.dart [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+        print("trustline for $currencyCode was added successfully");
+        return true;
+      }
+    }
+
+    print("No trustlines were processed");
+    return false;
+  }
+
+  Future<bool> sendTransaction(
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:152]:
```suggestion
  Future<bool> transfer(
```

File: packages/stellar_client/lib/src/client.dart [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+      await _sdk.submitTransaction(transaction);
+      print("Transaction successful.");
+      return true;
+    } catch (error) {
+      print("Failed to send transaction: $error");
+      return false;
+    }
+  }
+
+  Future<bool> activateThroughWallet({required String secretSeed}) async {
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:193]:
```suggestion
  Future<bool> activateThroughActivationWallet({required String activationWalletSecretSeed}) async {
```

File: packages/stellar_client/lib/src/client.dart [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+    try {
+      KeyPair walletKeyPair = KeyPair.fromSecretSeed(secretSeed);
+      AccountResponse existingAccount =
+          await _sdk.accounts.account(walletKeyPair.accountId);
+
+      Transaction transaction = TransactionBuilder(existingAccount)
+          .addOperation(CreateAccountOperationBuilder(accountId, "10").build())
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:200]: 10 is a bit much. AFAIR we were used to transfer 2.6 XLM to activate the wallet. Can you try that or less? we should use the minimal value

File: packages/stellar_client/lib/src/client.dart [hunk 8]
```diff
@@ -0,0 +1,440 @@
...
+  Future<bool> addTrustLine() async {
+    for (var entry in _currencies.currencies.entries) {
+      String currencyCode = entry.key;
+      currency.Currency Currency = entry.value;
+
+      String issuerAccountId = Currency.issuer;
+      Asset currencyAsset =
+          AssetTypeCreditAlphaNum4(Currency.assetCode, issuerAccountId);
+
+      // TODO:
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:125]: please remove this if there is nothing to be done

File: packages/stellar_client/lib/src/client.dart [hunk 9]
```diff
@@ -0,0 +1,440 @@
...
+
+      await _sdk.submitTransaction(transaction);
+      return true;
+    } catch (error) {
+      print("Failed to create account $error");
+      return false;
+    }
+  }
+
+  // take asset code code && issuer (optional)
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:213]:
```suggestion
  // take asset code && issuer (optional)
```

File: packages/stellar_client/lib/src/client.dart [hunk 10]
```diff
@@ -0,0 +1,440 @@
...
+        print('Failed to details for asset: ');
+      }
+
+      return null;
+    } catch (error) {
+      throw Exception("Failed to get fund details, $error");
+    }
+  }
+
+  Future<Operation?> makeFundPaumentOperation(
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:288]:
```suggestion
  Future<Operation?> makeFundPaymentOperation(
```

File: packages/stellar_client/lib/src/client.dart [hunk 11]
```diff
@@ -0,0 +1,440 @@
...
+        await fetchFundDetails(assetCode: assetCode);
+    Asset asset = AssetTypeCreditAlphaNum4(assetCode, issuer);
+
+    return PaymentOperationBuilder(
+            transactionData!.feeAccountId, asset, transactionData.feeFixed)
+        .build();
+  }
+
+  Future<Transaction?> buildFundedPaymentTransaction(
+      {required KeyPair sourceKeyPair,
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:300]: what is this source keypair used for?

File: packages/stellar_client/lib/src/client.dart [hunk 12]
```diff
@@ -0,0 +1,440 @@
...
+    TransactionData? transactionData =
+        await fetchFundDetails(assetCode: assetCode);
+    Asset asset = AssetTypeCreditAlphaNum4(assetCode, issuer);
+
+    return PaymentOperationBuilder(
+            transactionData!.feeAccountId, asset, transactionData.feeFixed)
+        .build();
+  }
+
+  Future<Transaction?> buildFundedPaymentTransaction(
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:299]: it's doing the same as sendTransaction(transfer) method but without sending the transaction. so can we reuse it in the sendTransaction(transfer) method?

File: packages/stellar_client/lib/src/client.dart [hunk 13]
```diff
@@ -0,0 +1,440 @@
...
+      await _sdk.submitTransaction(transaction);
+      return true;
+    } catch (error) {
+      print("Failed to create account $error");
+      return false;
+    }
+  }
+
+  // take asset code code && issuer (optional)
+  Future<bool> fundTrustLine(String asset_code) async {
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:214]:
```suggestion
  Future<bool> addTrustLineThroughThreefoldService(String asset_code) async {
```

File: packages/stellar_client/lib/src/client.dart [hunk 14]
```diff
@@ -0,0 +1,440 @@
...
+        .addOperation(paymentOperation!)
+        .addOperation(
+            PaymentOperationBuilder(destinationAddress, tftAsset, amount)
+                .build())
+        .addMemo(memoText != null ? Memo.text(memoText) : Memo.none())
+        .build();
+    return transaction;
+  }
+
+  Future<void> submitFundedTransaction(Transaction fundedTransaction) async {
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:338]: is it sending a funded transaction or funding the transaction and sending it?

File: packages/stellar_client/lib/src/client.dart [hunk 15]
```diff
@@ -0,0 +1,440 @@
...
+      account: response.account!,
+      startingBalance: double.parse(response.startingBalance!),
+    );
+  }
+
+  Future<List<Map<String, dynamic>>> getBalance() async {
+    List<Map<String, dynamic>> balancesList = [];
+    AccountResponse account = await _sdk.accounts.account(accountId);
+    for (Balance balance in account.balances) {
+      Map<String, dynamic> balanceMap = {};
```
Comment (AhmedHanafy725) [inline packages/stellar_client/lib/src/client.dart:426]: we already know the type here that should have asset code and balance. can we create an interface for it?

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
please make sure that the helper methods are private

File: packages/stellar_client/lib/src/client.dart (hunk 14 above)
Comment (AlaaElattar) [reply packages/stellar_client/lib/src/client.dart:338]: Funds and signs a TFT transaction.

File: packages/stellar_client/README.md
Comment (AhmedHanafy725) [inline packages/stellar_client/README.md:1]: can you please update the examples here?

File: packages/stellar_client/README.md [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+
+## Usage
+
+This client enables you to create new Stellar accounts or load existing ones on the testnet or public network. Once an account is created or loaded, you can manage assets such as TFT and XLM.
+
+### TESTNET
+
+Create new testnet account:
+
+- If no mnemonic provided, it will be created by the client.
```
Comment (AhmedHanafy725) [inline packages/stellar_client/README.md:21]: it depends on which method is called

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**